itl/command-plugins.conf

```diff
@@ -549,6 +549,195 @@ object CheckCommand "http" {
 	vars.http_verbose = false
 }
 
+object CheckCommand "curl" {
+	import "ipv4-or-ipv6"
+
+	command = [ PluginDir + "/check_curl" ]
+
+	arguments += {
+		"-H" = {
+			value = "$curl_vhost$"
+			description = "Host name argument for servers using host headers (virtual host). Append a port to include it in the header (eg: example.com:5000)"
+		}
+		"-I" = {
+			value = "$curl_ip$"
+			description = "IP address or name (use numeric address if possible to bypass DNS lookup)."
+		}
+		"-p" = {
+			value = "$curl_port$"
+			description = "Port number (default: 80)"
+		}
+		"-4" = {
+			set_if = "$curl_ipv4$"
+			description = "Force `check_curl` to use IPv4 instead of choosing automatically"
+		}
+		"-6" = {
+			set_if = "$curl_ipv6$"
+			description = "Force `check_curl` to use IPv6 instead of choosing automatically"
+		}
+		"-S" = {
+			set_if = "$curl_tls$"
+			value = "$curl_tls_version$"
```

Comment:
Tried it and I think this does not work as intented, since `"-S"` is the key of the array, so the second one overwrites the first one.

Reply:
Damn. I just tried some options, also with macros like the following, but they all failed.

While `value` allows to be a function, it will also result in being an empty string if this function returns null.

With `skip_key` being of type Boolean, this disallows having a macro to switch key and value for this, like:
```
value = {{ macro("$curl_tls_version$") ? macro("$curl_tls_version$") : "-S" }}
skip_key = {{ string(macro("$curl_tls_version$")) == "" }}
```

For the moment, I am out of ideas. Thus, I would suggest something like the following and fixing the quirky `-S` behavior upstream in the monitoring plugins.

```
    "-S" = {
      set_if = "$curl_tls$"
      value = {{ macro("$curl_tls_version$") ? macro("$curl_tls_version$") : null }}
      description = "Connect via SSL. Port defaults to 443. VERSION is optional, and prevents auto-negotiation"
    }
```

Reply:
Just had another idea after talking about this with @yhabteab. It is possible to overwrite the `key` attribute. The following CheckCommand arguments do work:

```
    "(-S w/ value)" = {
      set_if = {{ macro("$curl_tls$") && string(macro("$curl_tls_version$")) != "" }}
      key = "-S"
      value = "$curl_tls_version$"
      description = "Connect via SSL. Port defaults to 443. VERSION is optional, and prevents auto-negotiation"
    }
    "(-S w/o value)" = {
      set_if = {{ macro("$curl_tls$") && string(macro("$curl_tls_version$")) == "" }}
      key = "-S"
      description = "Connect via SSL. Port defaults to 443. VERSION is optional, and prevents auto-negotiation"
    }
```